Greedy to web marius

MariusKurbakovas opened the pull request:
Big pull request. Don't even know if it will work on your computer untill I try atleast on one. Today (2017-10-17) I will explain at least the basics of how everything works and how to continue adding new stuff.

MariusKurbakovas committed:
Web project template

MariusKurbakovas committed:
Checked if bootstrap works (It does)

MariusKurbakovas committed:
Front end deveopment, made some sort of carcass for our program. Made an example how to connect front end to back end, changed some config files a bit for easier programming, added logo, needed packages

MariusKurbakovas committed:
Made the main project greedy a console server, more progress on the front end, able to upload images and OCR them (still unable to show item list in frontend), UserLogin form (not connected to BackEnd), Other minor fixes

MariusKurbakovas committed:
Web now displays items from our default receipt, basic fixes, deleted uneeded classes and files

MariusKurbakovas committed:
Added bootstrap, some css to make the design more responsive and pleasurable, some frontend exception catching, code formating, deleting unneccesary files. Ready for a pull request

MariusKurbakovas committed:
Deleted one test class, because the class that it tests was deleted

MariusKurbakovas committed:
Minor fixes after looking at my code on github (left some trash in code and files)

valentk777 commented:
firstly, this is big job and nice to imagine a possible web version, but to merge this to master it takes some time to understand. First question: GREEDY SERVER (load failed). May someone can help to solve it?

MariusKurbakovas reviewed (commented):
I can try to solve it Valentinas, but first will need to see the error messages and logs. Maybe today before or during the C# lecture.

MariusKurbakovas committed:
Included one web pack without which server doesn't load

Commit:
decimal showing issue fixed

valentk777 commented:
i fixed decimal issue, so up to date your program :)

MariusKurbakovas committed:
User login (Both back end and front end): Users read from file, passwords being hashed, for now frontend deals with login pporly (no session, decides if a user is logged in only by one cookie named username). Added (only half working for now) facebook login button

MariusKurbakovas committed:
Merge branch 'GreedyToWebMarius' of https://github.com/GREEDY2/GREEDY into GreedyToWebMarius

MariusKurbakovas committed:
Checking if the test passes in AppVeyron (decimal number issue)

MariusKurbakovas reviewed (commented):
The check passes now. Thanks Valentinai

MariusKurbakovas committed:
Changed User Authentication Service to use LINQ

MariusKurbakovas committed:
Registration form (only front end) not finished.

MariusKurbakovas committed:
UserRegistration frontend finished, UserRegistration backend finished. If you wan't to login to app then username: admin, password: admin

MariusKurbakovas committed:
Item Editting Front End (Still in progress)

valentk777 commented on GREEDY.sln:
```diff
@@ -1,31 +1,37 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26730.16
+VisualStudioVersion = 15.0.26730.12
```
why we back to 12?

valentk777 commented on GREEDY/App.config:
```diff
@@ -1,21 +1,28 @@
+    <add key="CategoriesDataPath" value="../../Categories.json" />
+    <add key="UsersDataPath" value="../../Users.json" />
+    <add key="ServerAdressAndPort" value="http://localhost:6967/" />
+    <add key="EncryptionKey" value="marius" />
```
Is it okay to have EncryptionKey value like "marius"?

MariusKurbakovas replied:
Probably would be better randomly generated

valentk777 commented on GREEDY/Controllers/AuthenticationController.cs:
```diff
@@ -0,0 +1,56 @@
+                user.SessionId = session.SessionId;
+                return
+            }*/
+            return HelperClass.JsonHttpResponse<Object>(null);
```
you should save this code somewhere in your personal storage, but do not leave it in master.

valentk777 commented on GREEDY/Controllers/AuthenticationController.cs:
```diff
@@ -0,0 +1,56 @@
+            HttpContent requestContent = Request.Content;
+            string jsonContent = requestContent.ReadAsStringAsync().Result;
+            LoginCredentials credentials = JsonConvert.DeserializeObject<LoginCredentials>(jsonContent);
+            if (credentials.Username.Length < 1 || credentials.Password.Length < 1
```
Everything is good, but I strongly recommend to add security rules in this area, such that credentials should be no less than 6 or 8 symbols (not <1 as now) or is credentials.Password contains ONLY small letters and others. Possibly we can add this rules in one boolean method and in this "if" call that method to get the answer if everything correct

MariusKurbakovas replied:
Agree with the simbol count. Don't agree with only small letters (Like where have you seen that password is limited to some kind of characters :D)

valentk777 replied:
it was my mistake talking about ONLY small letter. It was just example about rules for password. we can create a rules like: more than 6 symbols, Upper case and lower case and number should be included. and give this information for the user. But now it's okay to do not have additional problems with security. but latter, we have to add this rules too

valentk777 commented on GREEDY/Controllers/RegistrationController.cs:
```diff
@@ -0,0 +1,40 @@
+            if (credentials.Username.Length < 1 || credentials.Password.Length < 1
+                || credentials.Username.Length > 256 || credentials.Password.Length > 256
+                || credentials.Email.Length < 3 || credentials.Email.Length > 256
+                || credentials.Fullname.Length < 3 || credentials.Fullname.Length > 513)
```
same as in Authentication. Need to think about security. and email length could be more than 5 (symbol@symbol.symbol)

MariusKurbakovas replied:
Actualy a@b is a valid email. But agree that the checking magament could have it's own class

valentk777 commented on GREEDY/DataManagers/DataConverter.cs:
```diff
@@ -19,6 +19,10 @@ public List<Item> ReceiptToItemList(Receipt receipt)
             var receiptLinesToString = String.Join(Environment.NewLine, receipt.LinesOfText);
             List<Item> itemList = new List<Item>();
 
+            System.Globalization.CultureInfo customCulture = (System.Globalization.CultureInfo)System.Threading.Thread.CurrentThread.CurrentCulture.Clone();
```
can we write this line shorter?

valentk777 commented on GREEDY/DataManagers/DataConverter.cs:
```diff
@@ -19,6 +19,10 @@ public List<Item> ReceiptToItemList(Receipt receipt)
+            customCulture.NumberFormat.NumberDecimalSeparator = ",";
+            System.Threading.Thread.CurrentThread.CurrentCulture = customCulture;
+
             if (shop == "RIMI" || shop == "MAXIMA")
```
need to add TODO: change strings to dictionary

valentk777 commented on GREEDY/DataManagers/UserRegistration.cs:
```diff
@@ -0,0 +1,57 @@
+            List<User> users = null;
+            if (!File.Exists(Environments.AppConfig.UsersDataPath))
+            {
+                //File.Create(Environments.AppConfig.UsersDataPath);
```
possible this week we will have database. so, almost all method will be changed there. Okay for this push

valentk777 commented on GREEDY/Extensions/StringEncryptExtension.cs:
```diff
@@ -0,0 +1,66 @@
+            using (Aes encryptor = Aes.Create())
+            {
+                Rfc2898DeriveBytes pdb = new Rfc2898DeriveBytes(EncryptionKey, new byte[] 
+                    { 0x49, 0x76, 0x61, 0x6e, 0x20, 0x4d, 0x65, 0x64, 0x76, 0x65, 0x64, 0x65, 0x76 });
```
magic numbers. need to add comment about this line

MariusKurbakovas replied:
Yeah a comment could really help, but I don't think there is a smart way to store these numbers somewhere

valentk777 commented on GREEDY/Extensions/StringEncryptExtension.cs:
```diff
@@ -0,0 +1,66 @@
+                using (Aes encryptor = Aes.Create())
+                {
+                    Rfc2898DeriveBytes pdb = new Rfc2898DeriveBytes(EncryptionKey, new byte[] 
+                        { 0x49, 0x76, 0x61, 0x6e, 0x20, 0x4d, 0x65, 0x64, 0x76, 0x65, 0x64, 0x65, 0x76 });
```
magic numbers

valentk777 commented on GREEDY/Services/AuthService.cs:
```diff
@@ -0,0 +1,78 @@
+            if (!File.Exists(Environments.AppConfig.UsersDataPath))
+            {
+                //File.Create(Environments.AppConfig.UsersDataPath);
+                // TO DO:Fix problem with file creation
```
TODO is one word :)

valentk777 commented on GREEDY/Services/AuthService.cs:
```diff
@@ -0,0 +1,78 @@
+        }
+        //TODO: Add logic for geting users session ID
+
+        /*public static string GetRoleBySessionId(AppDbContext _context, string sessionId)
```
need to save this commented code part in your local environment and do not push it to master

MariusKurbakovas replied:
:(

valentk777 commented on GREEDY_WEB/ClientApp/boot.tsx:
```diff
@@ -0,0 +1,30 @@
+import { AppContainer } from 'react-hot-loader';
+import { BrowserRouter } from 'react-router-dom';
+import * as RoutesModule from './routes';
+let routes = RoutesModule.routes;
```
A need a one line space there to see variable routes declaration easier

valentk777 commented on GREEDY_WEB/ClientApp/boot.tsx:
```diff
@@ -0,0 +1,30 @@
+import * as RoutesModule from './routes';
+let routes = RoutesModule.routes;
+
+function renderApp() {
```
you can change this line like 
"const renderApp = () => {"

valentk777 commented on GREEDY_WEB/ClientApp/components/Authorization.tsx:
```diff
@@ -0,0 +1,34 @@
+            <Route exact path='/forgot' component={ForgotPassword} />
+            <Route path='/' component={UserLogin} />
+        </Switch>
+    }
```
I suggest you to change alignment like in XML and return all return code inside ( ) brackets. 
it would be better to read.

valentk777 commented on GREEDY_WEB/ClientApp/components/Counter.tsx:
```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+
+interface CounterState {
```
ICounterState look better because this is our coding style :)

MariusKurbakovas replied:
This class will be deleted completely (it is still left from creating the project). Actually I only interface Props and State (and I don't call them iState and iProps, hope thats not too bad)

valentk777 commented on GREEDY_WEB/ClientApp/components/Counter.tsx:
```diff
@@ -0,0 +1,31 @@
+
+            <button className="btn btn-info" onClick={ () => { this.incrementCounter() } }>Increment</button>
+        </div>;
+    }
```
add all return code in to brackets ()

valentk777 commented on GREEDY_WEB/ClientApp/components/Counter.tsx:
```diff
@@ -0,0 +1,31 @@
+        return <div>
+            <h1>Counter</h1>
+
+            <p>This is a simple example of a React component.</p>
```
does we need this line?

valentk777 commented on GREEDY_WEB/ClientApp/components/Counter.tsx:
```diff
@@ -0,0 +1,31 @@
+        super();
+        this.state = { currentCount: 0 };
+    }
+
```
you can write it shorter, like 
      state = { currentCount: 0 };
without constructor(), super() and this

valentk777 replied:
https://michalzalecki.com/react-components-and-class-properties/

valentk777 commented on GREEDY_WEB/ClientApp/components/Counter.tsx:
```diff
@@ -0,0 +1,31 @@
+            currentCount: this.state.currentCount + 1
+        });
+    }
+}
```
change this code like:
    incrementCounter = () => {
        this.setState((prevState) => ({
            currentCount: prevState.currentCount + 1
        }));
    };

and add it in class. after that change 22 line to like this:
button className="btn btn-info" onClick={this.incrementCounter}>Increment</button>

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+    eItemPrice: number;
+    eItemCategory: string;
+    eSuccess: boolean;
+    
```
empty line should be deleted

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+import { ModalContainer, ModalDialog } from 'react-modal-dialog';
+import axios from 'axios';
+
+interface Props {
```
change name to IProps for your coding style standard

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+                let res = response.data;
+                if (res) {
+                    this.setState({ eSuccess: true, eHappened: true, showEdit: false });
+                    //this.update();
```
delete comment and add "TODO: implement this.update()" instead of commented code

MariusKurbakovas replied:
Ok thanks

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+                            <option>gerimai</option>
+                            <option>uztatas</option>
+                            <option>4</option>
+                            <option>5</option>
```
need to change magic numbers as 4 and 5 to something more useful. and all style is a bit strange, because we cannot named all possible categories if eItemCategory do not pass correctly. So, need to have a dictionary or smth like this with all default categories (if user do not have his category or our program fails setting category). on the other hand, possible this line will be changed when we will have database

MariusKurbakovas replied:
Yeah this is just the base model. In the place of the hardcoded numbers there will be the list of all our know categories + an option to add a new one.
When I coded this we didn't have a database, so I left it for later, but left everything as it is so you can see how everything works

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+import { Button, ButtonGroup, InputGroup, InputGroupAddon, Input, Form, FormGroup, Label, FormText } from 'reactstrap';
+import { ModalContainer, ModalDialog } from 'react-modal-dialog';
+
+interface Props {
```
change to IProps

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+    itemList: any
+}
+
+interface State {
```
change to IState

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+export class FetchData extends React.Component<Props, State> {
+    constructor() {
+        super();
+        /*axios.get('http://localhost:6967/api/ItemData')
```
save this commented area in you working branch and do not push to master :)

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+        this.setState({ showItems: true, showEdit: false, eHappened: false, itemList: this.props.itemList });
+    }
+
+
```
delete extra line

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    getItems() {
+        /*axios.get('http://localhost:6967/api/ItemData')
```
same with this code. need to delete it

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+    eCategoryChange(event) {
+        this.setState({ eItemCategory: event.target.value });
+    }
+
```
line 89-100 in EditItem.tsx

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+                    : null
+            }
+
+
```
delete one line

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+                </ModalContainer>
+            }
+        </div>;
+    }
```
140-170 same as in EditItem.tsx. can we joint it somehow?

valentk777 commented on GREEDY_WEB/ClientApp/components/ForgotPassword.tsx:
```diff
@@ -0,0 +1,17 @@
+﻿import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+
+
```
delete extra line

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+
+    public render() {
+        return <div>
+            {
```
all return add in brackets ( ... )

valentk777 commented on GREEDY_WEB/ClientApp/components/ForgotPassword.tsx:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    public render() {
+        return <div>
```
all return add in brackets ( ... )

valentk777 commented on GREEDY_WEB/ClientApp/components/Home.tsx:
```diff
@@ -0,0 +1,37 @@
+import { FetchData } from './FetchData';
+
+interface State {
+    itemList: any
```
name it as IState. Same interface as in FechData.tsx but then it is named as Props. Can we join them? or react does not work like c#?

valentk777 commented on GREEDY_WEB/ClientApp/components/Home.tsx:
```diff
@@ -0,0 +1,37 @@
+        this.setState({ itemList: items });
+    }
+
+    public render() {
```
all returned code in brackets ( )

valentk777 replied:
check enlightenment too

valentk777 commented on GREEDY_WEB/ClientApp/components/ImageUpload.tsx:
```diff
@@ -0,0 +1,49 @@
+        this.state = {
+            itemList: []
+        };
+    }
```
line from 10 to 15 change to short one     state = { itemList: [] };

valentk777 commented on GREEDY_WEB/ClientApp/components/Home.tsx:
```diff
@@ -0,0 +1,37 @@
+        this.getItemList = this.getItemList.bind(this);
+        this.state = {
+            itemList: []
+        };
```
change all state declaration to shorter like "state = { itemList: [] };" and delete constructor part

MariusKurbakovas replied:
The constructor needs to bind the method getItemList, so I don't think deleting it would work

valentk777 commented on GREEDY_WEB/ClientApp/components/ImageUpload.tsx:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    render() {
+        return (
```
there you added bracket for return, so as I mention, in every return should be brackets too :)

valentk777 commented on GREEDY_WEB/ClientApp/components/Layout.tsx:
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { NavMenu } from './NavMenu';
+
+export interface LayoutProps {
```
named it as ILayoutProps

valentk777 commented on GREEDY_WEB/ClientApp/components/Layout.tsx:
```diff
@@ -0,0 +1,21 @@
+
+export class Layout extends React.Component<LayoutProps, {}> {
+    public render() {
+        return <div className='container-fluid'>
```
need brackets for return code

valentk777 commented on GREEDY_WEB/ClientApp/components/NavMenu.tsx:
```diff
@@ -0,0 +1,107 @@
+import axios from 'axios';
+
+interface State {
+    loggedIn: boolean;
```
named it as IState

valentk777 commented on GREEDY_WEB/ClientApp/components/Authorization.tsx:
```diff
@@ -0,0 +1,34 @@
+    constructor(props) {
+        super(props);
+    }
+
```
why we need this constructor?

MariusKurbakovas replied:
Because this class takes in arguments from another class (the arguments are called props) and I believe in react if you want to reference those arguments correctly they need to be passed to the base react class (the one this extends from). And that means you have a constructor that calls its father constructor

valentk777 replied:
okay, thanks :)

valentk777 commented on GREEDY_WEB/ClientApp/components/Authorization.tsx:
```diff
@@ -0,0 +1,34 @@
+        super(props);
+    }
+
+    componentUpdate() {
```
recommend you to change method to property like:     componentUpdate = () => {

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+export class EditItem extends React.Component<Props> {
+    constructor(props) {
+        super(props);
+    }
```
why we need this constructor. what does it do?

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+            });
+    }
+
+    eNameChange(event) {
```
suggesting you to change it to property, because it's only set value
eNameChange = (event) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+        this.setState({ eItemName: event.target.value });
+    }
+
+    ePriceChange(event) {
```
suggesting you to change it to property, because it's only set value
ePriceChange = (event) =>  {

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+        this.setState({ eItemPrice: event.target.value });
+    }
+
+    eCategoryChange(event) {
```
suggesting you to change it to property, because it's only set value
eCategoryChange = (event) =>  {

valentk777 commented on GREEDY_WEB/ClientApp/components/EditItem.tsx:
```diff
@@ -0,0 +1,87 @@
+    }
+
+    public render() {
+        return <ModalContainer onClose={this.props.hideEdit}>
```
need to add brackets ( )

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+        this.saveItemChanges = this.saveItemChanges.bind(this);
+        this.eCategoryChange = this.eCategoryChange.bind(this);
+        this.eNameChange = this.eNameChange.bind(this);
+        this.ePriceChange = this.ePriceChange.bind(this);
```
You can remove all this line and change methods to property like 
    update = () => {
        this.getItems();
    }

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+    }
+
+
+    update() {
```
update = () => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+        this.getItems();
+    }
+
+    getItems() {
```
getItems = () => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+            });*/
+    }
+
+    editItem(id, name, price, category) {
```
editItem = (id, name, price, category) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+        this.setState({ showEdit: true, eItemId: id, eItemName: name, eItemCategory: category, eItemPrice: price });
+    }
+
+    hideEdit() {
```
hideEdit = () => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+        this.setState({ showEdit: false });
+    }
+
+    saveItemChanges(e) {
```
saveItemChanges = (e) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+            });
+    }
+
+    eNameChange(event) {
```
eNameChange = (event) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+        this.setState({ eItemName: event.target.value });
+    }
+
+    ePriceChange(event) {
```
ePriceChange = (event) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+        this.setState({ eItemPrice: event.target.value });
+    }
+
+    eCategoryChange(event) {
```
eCategoryChange = (event) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/FetchData.tsx:
```diff
@@ -0,0 +1,171 @@
+                                    <option>{this.state.eItemCategory}</option>
+                                    <option>gerimai</option>
+                                    <option>uztatas</option>
+                                    <option>4</option>
```
came comment as in EditItem.tsx

valentk777 commented on GREEDY_WEB/ClientApp/components/Home.tsx:
```diff
@@ -0,0 +1,37 @@
+        };
+    }
+
+    public getItemList(items) {
```
change this to     
public getItemList = (items) => { 
and then delete declaration in constructor

valentk777 commented on GREEDY_WEB/ClientApp/components/ImageUpload.tsx:
```diff
@@ -0,0 +1,49 @@
+import ImageUploader from 'react-images-upload';
+import axios from 'axios';
+
+interface Props {
```
named it as IProps

valentk777 commented on GREEDY_WEB/ClientApp/components/ImageUpload.tsx:
```diff
@@ -0,0 +1,49 @@
+
+        this.state = {
+            itemList: []
+        };
```
why we need this?

MariusKurbakovas replied:
If I remeber correctly this is because of typescript. Javascript lets you do anything with undefined variables but typescript show an error. So I declered it (even if I declared it as an empty array it is still defined)

valentk777 commented on GREEDY_WEB/ClientApp/components/NavMenu.tsx:
```diff
@@ -0,0 +1,107 @@
+    loggedIn: boolean;
+    username: string;
+}
+
```
IState is not used in code

valentk777 commented on GREEDY_WEB/ClientApp/components/NavMenu.tsx:
```diff
@@ -0,0 +1,107 @@
+
+export class NavMenu extends React.Component<{}, {}> {
+    timer: any;
+    constructor(props) {
```
delete all constructor and change it to only 
    state = { loggedIn: false }

valentk777 commented on GREEDY_WEB/ClientApp/components/NavMenu.tsx:
```diff
@@ -0,0 +1,107 @@
+        this.timer = setInterval(() => this.checkIfLoggedIn(), 100);
+    }
+
+    checkIfLoggedIn() {
```
checkIfLoggedIn = () => {

valentk777 commented on GREEDY_WEB/ClientApp/components/NavMenu.tsx:
```diff
@@ -0,0 +1,107 @@
+        clearInterval(this.timer);
+    }
+
+    handleLogout() {
```
handleLogout = () => {

valentk777 commented on GREEDY_WEB/ClientApp/components/NavMenu.tsx:
```diff
@@ -0,0 +1,107 @@
+        const cookies = new Cookies();
+        cookies.remove('username', { path: '/' });
+        //TODO: clear session once we have one
+        /*axios.post(`/api/Authentication/Logout`)
```
Clear all commented code

valentk777 commented on GREEDY_WEB/ClientApp/components/NavMenu.tsx:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    public render() {
+        return <div className='main-nav'>
```
return in bracket ( .. )

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+import FacebookLogin from 'react-facebook-login';
+
+export class UserLogin extends React.Component<RouteComponentProps<{}>> {
+    constructor(props) {
```
delete constructor

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+        this.onFormSubmit = this.onFormSubmit.bind(this);
+    }
+
+    onFormSubmit(e, next) {
```
onFormSubmit = (e, next) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+        e.preventDefault();
+        var data = e.data;
+
+        if (data.username.length < 1) {
```
we have to have 1 as a constant because if we cant to change it as 5 (for instance) for security purpose, then we will have to change it in GREEDY/Controllers/RegistrationController.cs too. also error message will be changed too

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+        }
+
+        // Require passwords to be at least 10 characters.
+        if (data.password.length < 1) {
```
same as in username. we have to have 1 as a constant and for security purpose we need a longer password too, Also, we need to add more if for upper and lower case and etc. I comment to have additional method for password verification.

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+                if (res) {
+                    const cookies = new Cookies();
+                    cookies.set('username', res.Username, { path: '/' });
+                    /* set sessionId for user
```
delete commented lines

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+    }
+
+    public render() {
+        return <div>
```
return code in brackets

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+                                                </p>
+                                                <Button className="col-xs-12 col-sm-12" type="submit" color="btn btn-primary buttonText">Login</Button>
+                                                <Link to="/registration" className="pull-right marginTop loginLinkButton">Don't have an account? Register</Link>
+                                                { /* <Link to="/forgot" className="pull-right">Forgot Password</Link> */
```
delete commented lines

valentk777 commented on GREEDY_WEB/ClientApp/components/UserLogin.tsx:
```diff
@@ -0,0 +1,116 @@
+            </DocumentTitle>
+        </div>;
+    }
+}
```
all this lines is too long to read. need to split it

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+import Cookies from 'universal-cookie';
+import FacebookLogin from 'react-facebook-login';
+
+interface State {
```
IState

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+}
+
+export class UserRegistration extends React.Component<RouteComponentProps<{}>, State> {
+    constructor(props) {
```
Delete constructor

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+        this.onFormSubmit = this.onFormSubmit.bind(this);
+        this.state = {
+            isAccountCreated: false
+        }
```
state = { isAccountCreated: false }

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+        }
+    }
+
+    onFormSubmit(e, next) {
```
onFormSubmit = (e, next) => {

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+        e.preventDefault();
+        var data = e.data;
+
+        if (data.username.length < 1) {
```
same constant as verifying username and same modified error message

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+            return next(new Error('Username must not be empty.'));
+        }
+
+        if (data.email.length < 1) {
```
recommended to have this "1" as constant

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+            return next(new Error('Email is too long.'));
+        }
+
+        // Require passwords to be at least 5 characters.
```
your comment in userLogin.tsx is
"        // Require passwords to be at least 10 characters."
that's why we need save values as constant

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+
+    public render() {
+
+
```
delete extra space

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+    public render() {
+
+
+        return <div>
```
brackets ( .. )

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+        return <div>
+            <DocumentTitle title={`Register`}>
+                <div className="container">
+                    {/*<div className="row">
```
remove commented lines

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+                                            Your account has been created.
+                                            <Link to="/" className="pull-right">Login Now</Link>.
+                                        </p>
+                                        {/*<p className="alert alert-success">Your account has been created. Please check your email for a verification link.</p>
```
same in there

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+                                        <div className="form-horizontal">
+                                            <div className="form-group">
+                                                <label htmlFor="spUsername" className="col-xs-12 col-sm-4 control-label">Username</label>
+                                                <div className="col-xs-12 col-sm-4">
```
lines is too long for reading

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+                                </div>
+                            }
+
+
```
extra lines should be removed

valentk777 commented on GREEDY_WEB/ClientApp/components/UserRegistration.tsx:
```diff
@@ -0,0 +1,203 @@
+                            </div >
+                        </div >
+                    </LoginForm > */}
+                </div >
```
remove commented text from possible master branch

valentk777 commented on GREEDY_WEB/Controllers/HomeController.cs:
```diff
@@ -0,0 +1,23 @@
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

remove unnecessary usings

valentk777 reviewed (changes requested):
most of my comments is a recommended, but not necessary to change. After this changes
_tesseract.GetReceipt(); extensions is not working correctly. need to find why we get this issue. for that reason I stop this branch and do not let to merge to master.

on the other hand, YOU made a hedge step! Thanks for that and keep on it. :)

Commit:
cosmetic changes related with comments in code review.

Commit:
still cosmetic changes

Commit:
Setup database, changed registration and login to work with database, after uploading photo items will be saved to database

Commit:
merging changes

valentk777 commented on GREEDY/Data/DataBaseModel.cs:
```diff
@@ -0,0 +1,23 @@
+        {
+            Database.SetInitializer<DataBaseModel>(new DropCreateDatabaseIfModelChanges<DataBaseModel>());
+        }
+
```
need to remove extra lines

MariusKurbakovas committed:
Logo got it's own component, sending username from FrontEnd to BackEnd when uploading an image. Controller fixed (no need to hardcode the username)

MariusKurbakovas committed:
Added Item Category Add FrontEnd (not connected to anything, only for showing purposes)

Commit:
Test Data added.
Two shops receipt reader implemented.
Picture filter frim Brigita's branche added.
Unit test fixed (still 2 not prepared for new reeipt format)

Commit:
cosmetic changes. only coding style changed, TODO added and some of that has been remmembered.

MariusKurbakovas reviewed (commented):
2 Unit Tests don't pass anymore Valentinas

valentk777 commented:
yes, I know. This is because we delete face receipt with ******* smth ***** and added regex for real one. so I will fix it later, because yesterday I do not have enough energy for that :D

MariusKurbakovas committed:
Fixed centering issues (both for phone and wide screen)

Commit:
Packages added

Commit:
Unit test pass now. need to implement more unit test and fix data converter method splitting it into parts

Commit:
final changes

valentk777 commented:
I think everything looks fine for now. Later needs to check why appveyor gets an error, but locally everything looks good
